gradle-plugin/src/main/groovy/mozilla/telemetry/glean-gradle-plugin/GleanGradlePlugin.groovy

```diff
@@ -0,0 +1,353 @@
+                // be extracted from the AAR file. The parser will just ignore this one, if not present.
+                args "${project.projectDir}/metrics.yaml"
+                args "${project.projectDir}/pings.yaml"
+
+                // If we're building the Glean library itself (rather than an
+                // application using Glean) pass the --allow-reserved flag so we can
+                // use metrics in the "glean..." category
+                if (project.ext.has("allowGleanInternal")) {
+                    args "--allow-reserved"
+                }
+
+                doFirst {
+                    // Add the potential 'metrics.yaml' files at evaluation-time, rather than
+                    // configuration-time. Otherwise the Gradle build will fail.
+                    if (project.ext.has("allowMetricsFromAAR")) {
+                        inputs.files.forEach { file ->
+                            project.logger.lifecycle("Glean SDK - generating API from ${file.path}")
+                            args file.path
+                        }
+                    }
+                }
+
+                // Only show the output if something went wrong.
+                ignoreExitValue = true
+                standardOutput = new ByteArrayOutputStream()
+                errorOutput = standardOutput
+                doLast {
+                    if (execResult.exitValue != 0) {
+                        throw new GradleException("Process '${commandLine}' finished with non-zero exit value ${execResult.exitValue}:\n\n${standardOutput.toString()}")
+                    }
+                }
+            }
+
+            // Only attach the generation task if the metrics file is available or we're requested
+            // to fetch them from AAR files. We don't need to fail hard otherwise, as some 3rd party
+            // project might just want metrics included in Glean and nothing more.
+            if (project.file("${project.projectDir}/metrics.yaml").exists()
+                || project.file("${project.projectDir}/pings.yaml").exists()
+                || project.ext.has("allowMetricsFromAAR")) {
+                // This is an Android-Gradle plugin 3+-ism.  Culted from reading the source,
+                // searching for "registerJavaGeneratingTask", and finding
+                // https://github.com/GoogleCloudPlatform/endpoints-framework-gradle-plugin/commit/2f2b91476fb1c6647791e2c6fe531a47615a1e85.
+                // The added directory doesn't appear in the paths listed by the
+                // `sourceSets` task, for reasons unknown.
+                variant.registerJavaGeneratingTask(generateKotlinAPI, new File(sourceOutputDir))
+            }
+        }
+    }
+
+    private Task generateGleanMetricsDocs
+
+    void setupGenerateMarkdownDocsTasks(Project project, File condaDir) {
+        // Generate the Metrics docs, if requested.
+        if (project.ext.has("gleanGenerateMarkdownDocs")) {
```

Comment:
I'm afraid this needs to be refreshed again :( the `sdk_generator.gradle` in this repo has a few updates (i.e. metrics from AARs for docs as well) that need to be ported over.

Reply:
All the other changes look good, but this still needs to be addressed, I think

Reply:
Yep.  I'm on it -- sorry to push before I was done ;)

Reply:
Woops, no, my bad for running through this before you had flagged me again :)